### PR TITLE
VPA - Reducing copy-pasted code in updater tests

### LINKED ISSUE #2793: VPA - Updater logic tests clean-up
[Updater logic test](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/updater/logic/updater_test.go) should be simplified. Copy-pasted code should be reduced.

### PR DESCRIPTION
**Pull request by krzysied**
/fixes #2793

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/updater/logic/updater_test.go [hunk 1]
```diff
@@ -209,11 +177,14 @@ func TestRunOnce_InvalidStatus(t *testing.T) {
 		recommendationProcessor:      &test.FakeRecommendationProcessor{},
 		selectorFetcher:              mockSelectorFetcher,
 		useAdmissionControllerStatus: true,
-		statusValidator:              newFakeValidator(false),
+		statusValidator:              statusValidator,
 	}
 
+	if expectFetchCalls {
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/updater/logic/updater_test.go]: Is there value from verifying those calls are not made if not expected?

Comment (krzysied) [reply vertical-pod-autoscaler/pkg/updater/logic/updater_test.go]: 1. This verification was in original tests, so I want to preserve the same logic.
2. IIUC using gomock controller defines a list of calls that we expect to happen. If the call is registered and not used it throws exception at the end of the test.
This is the first time I'm seeing this mock controller so I can be wrong here...

Comment (bskiba) [reply vertical-pod-autoscaler/pkg/updater/logic/updater_test.go]: I see, thanks

**Pull request merged**